source/set-mutable.d.ts

```diff
@@ -0,0 +1,37 @@
+import {Except} from './except';
+import {Mutable} from './mutable';
+
+/**
+Create a type that converts the given keys from `readonly` to mutable. The remaining keys are kept as is.
+
+Use-case: You want to define a single model where the only thing that changes is whether or not some of the keys are mutable.
+
+@see [`Mutable`](source/mutable.d.ts) - a type that sets all (not selectively) keys as mutable.
+
+@example
+```
+import {SetMutable} from 'type-fest';
+
+type Foo = {
+	readonly a: number;
+	b: readonly string[];
+	readonly c: boolean;
+}
+
+type SomeMutable = SetMutable<Foo, 'b' | 'c'>;
+// type SomeMutable = {
+// 	readonly a: number;
+// 	b: readonly string[]; // Was already mutable and still is, type of the property remains unaffected.
+// 	c: boolean; // Is now mutable.
+// }
+```
+*/
+export type SetMutable<BaseType, Keys extends keyof BaseType> =
+	// Pick just the keys that are not mutable from the base type.
+	Except<BaseType, Keys> &
+	// Pick the keys that should be mutable from the base type and make them mutable.
+	Mutable<Pick<BaseType, Keys>> extends
+	// If `InferredType` extends the previous, then for each key, use the inferred type key.
+	infer InferredType
+		? {[KeyType in keyof InferredType]: InferredType[KeyType]}
```

Comment:
What's the reason for doing this step rather than just doing:

```typescript
export type SetMutable<BaseType, Keys extends keyof BaseType> =
  // Pick just the keys that are not mutable from the base type.
  Except<BaseType, Keys> &
  // Pick the keys that should be mutable from the base type and make them mutable.
  Mutable<Pick<BaseType, Keys>>;
```

I guess the reason is that it makes it easier to read what the type actually is? As eg. VS Code Intellisense shows the actual keys and values rather than `Pick<Foo, "a"> & Mutable<Pick<Foo, "b" | "c">>`?

 don't know if this pattern is really something that's used throughout `type-fest` or whether it should be a pattern that's used throughout `type-fest`. What are your thoughts @sindresorhus?

The purist/perfectionist in me says that this is bad and just complicating the type setup to workaround a developer experience thing that should really be improved in TypeScript itself, although the convenient/pragmatist part of me says: This is great, so much more readable!

If we want to double down on this patter, which we for developer experience and consistency's sake should if it should remain, then creating a simple shared "simplifier" type would be preferable:

```typescript
type Simplify<T> = {[KeyType in keyof T]: T[KeyType]};
```

Using such a `Simplify<>` type, `SetMutable` could – while being functionally identical – be simplified to:

```typescript
export type SetMutable<BaseType, Keys extends keyof BaseType> =
  Simplify<
    // Pick just the keys that are not mutable from the base type.
    Except<BaseType, Keys> &
    // Pick the keys that should be mutable from the base type and make them mutable.
    Mutable<Pick<BaseType, Keys>>
  >;

Reply:
This implementation is really just an extension of the other `Set-` types. If it is decided that it should be changed I believe all of them should be updated.

Reply:
Right, another question for @sindresorhus then 👍

Reply:
> If we want to double down on this patter, which we for developer experience and consistency's sake should if it should remain, then creating a simple shared "simplifier" type would be preferable:

That sounds like a good idea. I think it's just the case of no one noticing the repetition/duplication.

Reply:
Applied.